Automattic/wp-calypso
# Display notices when introductory offers may apply or not during checkout

#### Changes proposed in this Pull Request

* When an introductory offer is part of the default product, display a different message in the interval selection during checkout.

![image](https://user-images.githubusercontent.com/15204776/141011720-00417b0e-3e0d-4e1e-82eb-c60464facae2.png)

* When an introductory offer is not applicable during checkout, display a notice in the line items section.

![image](https://user-images.githubusercontent.com/15204776/141011754-2555a457-4f83-4ece-89d8-e61ab3d3964a.png)

#### Testing instructions

* Depends on D69781-code
* Connect a new site using Jetpack
* Select a Jetpack Security yearly subscription
* When you edit the line items, verify that the savings message is displayed as in the screenshot from the description.
* Remove the Jetpack Security subscription and purchase it again.
* A notice in the line items section during checkout indicates that the order is not eligible for the discount.
